src/main/java/bio/terra/service/journal/JournalService.java

```diff
@@ -0,0 +1,208 @@
+package bio.terra.service.journal;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.model.JournalEntryModel;
+import bio.terra.service.auth.iam.IamResourceType;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JournalService {
+  private static final Logger logger = LoggerFactory.getLogger(JournalService.class);
+  private final ObjectMapper objectMapper;
+  private final JournalDao journalDao;
+
+  public JournalService(ObjectMapper objectMapper, JournalDao journalDao) {
+    this.objectMapper = objectMapper;
+    this.journalDao = journalDao;
+  }
+
+  /**
+   * Create a journal entry that indicates a resource was created
+   *
+   * @param user The user performing the action being journaled
+   * @param resourceKey The UUID of the resource (domain object) the journal entry pertains to
+   * @param resourceType The {@code IamResourceType} that maps to the domain object the journal
+   *     entry pertains to
+   * @param note An optional message describing the event
+   * @param changeMap An optional map of changes to store with the event
+   * @return UUID of the entry created.
+   */
+  public UUID journalCreate(
+      @NotNull AuthenticatedUserRequest user,
+      @NotNull UUID resourceKey,
+      @NotNull IamResourceType resourceType,
+      String note,
+      Map changeMap) {
```

Comment:
In general I think Java code should avoid use of generics without parameters. I think javac will generate a warning here, and sonar probably won't like it either. If the map contents are truly untyped, you can either use `Map<?, ?>`, which is the typed way to write the same thing as `Map`, or you could make each method parameterized on the K and V types for the map (or even better, something like `Map<String, Object>` if that works).

Reply:
I'm settling on Map<?, ?> for now.